src/Units/MoveTypes/MoveTypeFey.java

```diff
@@ -26,29 +24,21 @@ public MoveType clone()
   }
 
   @Override
-  public FloodFillFunctor getUnitMoveFunctor(Unit mover, boolean includeOccupied, boolean canTravelThroughEnemies)
+  public int getTransitionCost(GameMap map, XYCoord from, XYCoord to,
+                               Unit mover, boolean canTravelThroughEnemies)
   {
-    return new FeyMoveFillFunctor(mover, this, includeOccupied, canTravelThroughEnemies);
+    // Mandatory sanity check
+    if( !map.isLocationValid(to) )
+      return IMPASSABLE;
+
+    Environment endEnv = map.getEnvironment(to);
+    // Fey units cannot enter enemy-controlled spaces.
+    if( null != mover && mover.model.healableHabs.contains(endEnv.terrainType)
+        && endEnv.terrainType.isCapturable()
+        && mover.CO.isEnemy(map.getLocation(to).getOwner()) )
+      return IMPASSABLE;
```

Comment:
Do we need the `healableHabs` check? I would think the `isEnemy` check would be sufficient. Seems odd to ban units from entering a space just because it could heal them (if friendly).